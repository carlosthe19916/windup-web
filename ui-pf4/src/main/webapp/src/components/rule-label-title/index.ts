export { RulelabelTitle } from "./rule-label-title";
