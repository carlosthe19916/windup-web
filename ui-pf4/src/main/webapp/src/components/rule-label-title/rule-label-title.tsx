import React from "react";
import { Tooltip } from "@patternfly/react-core";
import {
  WarningTriangleIcon,
  ExclamationCircleIcon,
} from "@patternfly/react-icons";
import {
  global_warning_color_100 as warningColor,
  global_danger_color_100 as dangerColor,
} from "@patternfly/react-tokens";

export interface RulelabelTitleProps {
  name: string;
  errors: string[];
  numberOfRulesLabels: number;
}

export const RulelabelTitle: React.FC<RulelabelTitleProps> = ({
  name,
  errors,
  numberOfRulesLabels,
}) => {
  let info;
  if (errors.length > 0) {
    info = (
      <Tooltip content={errors.join(", ")}>
        <i>
          {" "}
          <ExclamationCircleIcon color={dangerColor.value} />
        </i>
      </Tooltip>
    );
  }
  if (numberOfRulesLabels === 0) {
    info = (
      <Tooltip content="Could not identify any rule inside this file/folder">
        <i>
          {" "}
          <WarningTriangleIcon color={warningColor.value} />
        </i>
      </Tooltip>
    );
  }

  return (
    <span>
      <span>{name}</span>
      {info}
    </span>
  );
};
