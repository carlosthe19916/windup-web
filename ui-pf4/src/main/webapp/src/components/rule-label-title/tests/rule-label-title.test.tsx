import React from "react";
import { shallow } from "enzyme";
import { RulelabelTitle } from "../rule-label-title";

describe("RulelabelTitle", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <RulelabelTitle name="My rule" errors={[]} numberOfRulesLabels={0} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
