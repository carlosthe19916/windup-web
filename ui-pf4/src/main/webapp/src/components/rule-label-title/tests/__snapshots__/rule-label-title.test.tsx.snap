// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RulelabelTitle Renders without crashing 1`] = `
<span>
  <span>
    My rule
  </span>
  <Tooltip
    content="Could not identify any rule inside this file/folder"
  >
    <i>
       
      <WarningTriangleIcon
        color="#f0ab00"
        noVerticalAlign={false}
        size="sm"
      />
    </i>
  </Tooltip>
</span>
`;
